WordPress/src/test/java/org/wordpress/android/ui/jetpackoverlay/individualplugin/WPJetpackIndividualPluginHelperTest.kt

```diff
@@ -46,8 +51,32 @@ class WPJetpackIndividualPluginHelperTest : BaseUnitTest() {
             assertThat(helper.shouldShowJetpackIndividualPluginOverlay()).isFalse
         }
 
+    @Suppress("MaxLineLength")
+    @Test
+    fun `GIVEN config is on, has problem sites, and shown more than max count WHEN shouldShowJetpackIndividualPluginOverlay THEN return false`() =
+        test {
+            whenever(wpIndividualPluginOverlayFeatureConfig.isEnabled()).thenReturn(true)
+            val connectedSites = listOf(
+                jetpackCPConnectedSiteModel(
+                    name = "site1",
+                    url = "site1.com",
+                    activeJpPlugins = "jetpack-social"
+                ),
+                jetpackCPConnectedSiteModel(
+                    name = "site2",
+                    url = "site2.com",
+                    activeJpPlugins = "other-plugin"
+                )
+            )
+            whenever(siteStore.getJetpackCPConnectedSites()).thenReturn(connectedSites)
+            whenever(appPrefs.wpJetpackIndividualPluginOverlayShownCount).thenReturn(3)
+
+            assertThat(helper.shouldShowJetpackIndividualPluginOverlay()).isFalse
+        }
+
+    @Suppress("MaxLineLength")
     @Test
-    fun `GIVEN config is on and has problem sites WHEN shouldShowJetpackIndividualPluginOverlay THEN return true`() =
+    fun `GIVEN config is on, has problem sites, and shown less than max count WHEN shouldShowJetpackIndividualPluginOverlay THEN return true`() =
```

Comment:
> **Note**
> This is not a request for change, just sharing my thoughts with you because I get the feeling you're also always looking for ways to improve existing code 🙇🏻 

I'm not very much sold on GIVEN-WHEN-THEN 😅 , but I am well aware that a lot of awesome engineers like you seem to prefer it.

To me something along more domain-specific feels cleaner and easier to read:
1. define a region in the test class for `Feature Flag Enabled`, or name it as you'd like
2. Write test names in it as such:
```
`on X event, when overlay shown less times than the limit, it is shown again`
```

Disclaimer: In my opinion tests should be readable by non-programmers, and to me the GIVEN-THEN-WHEN approach you took is very good as well.

PS: I gave up on these test naming patterns when I noticed a senior test automation engineer could not teach a medior manual tester -- which was very eager to learn test automation -- to write relevant test names. When I picked up the task to buddy the junior in their learning journey, I could only teach them when I tried different approaches to test naming. They ended up writing great test names and feeling confident in their work 😇.

Reply:
That's a great discussion topic! I tried a few different ways in the past as well. I ended up sticking to GIVEN-WHEN-THEN, or the sibling (but more technical) ARRANGE-ACT-ASSERT sometimes, due to it being easy to understand and articulate in order: CONDITIONS -> ACTIONS -> EXPECTED BEHAVIOR, so it's both easier for everyone to read it, and for developers to know where to find the part of the code referring to each segment of the test name.

Sometimes though I feel like writing it the way you suggested (action first, conditions second, expectations last) would still be very human readable and more concise, avoiding the "unneeded" words of the pattern.

My only nitpick is that both "patterns" use the word **when** in different ways and I fear this might be confusing to everyone contributing to the same codebase. In "GWT" when indicates the ACTION, while in your suggestion it is used to indicate the CONDITION.

All that said, I will definitely keep that in mind and try to write some tests using the pattern you suggested. I just discovered **regions** recently and will try to use it more.

Thanks again for the review and for raising this discussion!